### DISCUSSION TITLE
Exceptions are different when thrown in tests

### DISCUSSION THREAD

**Discussion by ash-jc-allen** [livewire/livewire#3048] [7. Bugs (archived)]
### Description
Hi! Apologies if I'm just barking up the wrong tree here and doing something wrong.

I have a component that is rendered inside a Blade view. In the ` mount() ` method of the component, I validate a field that is passed in. If the field is incorrect, I throw an ` \InvalidArgumentException `. This seems to work as expected.

But, if I try and do the same during testing, I seem to get a ` Facade\Ignition\Exceptions\ViewException ` instead. Is this just a limitation of the framework, or a bug, or am I approaching this wrong?

Cheers for any help in advance! :smile:

### Exact steps to reproduce

Livewire component (snippet):

```php
public function mount(): void
{
    if (! in_array($this->action, ['store', 'update'])) {
        throw new \InvalidArgumentException('The action is invalid.');
    }
}
```

How it's being called from the Blade view:
```
@livewire('store-update-records', ['action' => 'store'])
```

The test that is failing because the exception being thrown is a ` Facade\Ignition\Exceptions\ViewException `

```php
/** @test */
public function exception_is_thrown_if_the_action_is_invalid()
{
    $this->expectException(\InvalidArgumentException::class);
    $this->expectExceptionMessage('The action is invalid.');

    Livewire::test(StoreUpdateRecords::class, [
        'action' => 'INVALID',
    ]);
}
```

### Stripped-down, copy-pastable code snippets

**Context**
- Livewire version: v2.4.0
- Laravel version: v8.31.0
- Alpine version: -
- Browser: -

**Comment by austenc** [comment]
Thanks for reporting this @ash-jc-allen. I'll see if I can reproduce it and get back to you.

**Comment by titantwentyone** [comment]
Just came across this myself and wondered if there was a workaround? What I have found is that this happens in `mount` but if the exception is thrown in a method used as an action, the correct exception can be tested.

**Reply by ash-jc-allen** [reply]
I don't think I ever found a workaround for this. But I'm not working on that project anymore, so I can't double-check if it still exists,  sorry :)
